### ISSUE TITLE
RDR2

### ISSUE THREAD

**Issue by VR-Punk** [LukeRoss00/gta5-real-mod#28]
i have no idea about software development, so please excuse my naive question. Do you have any plans to start the development for RDR2, if the shaderhackers release 3D Vision support and the scripthook releases are more advanced ? Or should i better use vorpx for RDR2 with dx12 alpha support and fake3d, and should give up hope for a  r.e.a.l. mod vr in the in the near future ?

**Comment by spybeef** [comment]
What a brilliant idea, RDR2 with a VR mod like this one would be a dream come true.
In any case, really wanted to congratulate LureRoss00 on this GTA mod, it is absolutely breathtaking! Thank you much!!! (I am aware this is not an "issue", but couldn't find a better way to share my appreciation!c :)

**Comment by cercata** [comment]
https://github.com/LukeRoss00/gta5-real-mod/issues/18

**Comment by LukeRoss00** [comment]
Thanks for the feedback!

So, I just finished a few benchmarking sessions with RDR2, and the news is not good. With Balanced (default) settings, at the resolution and FOV needed for "real" VR, Red Dead Redemption 2 struggles to reach 40 fps on a system with i9-9900K, 32 GB dual channel, GTX 1080 Ti, and of course SSD.

To be clear, that's less than 40 _total_ frames per second, or less than 20 per eye, even with alternate eye rendering.

With Performance settings, i.e, basically everything on Low, it runs at around 60-70 total fps (30-35 per eye, still not enough), and it looks like... well, you can imagine.

Please don't take this to mean that the game is not "optimized" as kids on the forums love to say. At 1080p, it pushes more than 80 fps (in other terms more than needed) while looking gorgeous. But there is no way that during 2020 a new GPU will come out that will be able to render RDR2 at the combination of quality, resolution, FOV and frame rate that is needed for proper VR. With SLI completely dead and no longer supported by either games or VR runtimes, there is not even the possibility to throw heaps of money at the problem.

Additionally, my GTA V mod was built around 3Dmigoto, which is limited to DX11, whereas RDR2 surprisingly came out supporting only DX12 and Vulkan. Unless 3Dmigoto is extended to DX12 (extremely unlikely), or RDR2 is backported to DX11 (yeah, sure), the mod would need to be rewritten from scratch, together with a new hooking library.

TL;DR: Don't hold your breath for a proper VR mod that displays RDR2 in its full glory, the software and most importantly the hardware is just not there.

**Comment by VR-Punk** [comment]
I think we have to wait until the new nvidia or intel cards with 15-20 teraflops are payable next year. In the most games i set graphic settings in vorpx to medium but the textures on ultra anyway. If i use fake3d, then i will set the resolution to 2500x2500 or 3000x3000 and for geometric 3d 1500x1500 or 2000x2000. With my rtx 2080 i get 30-50 fps and using Asynchronous reprojection it will be upscaled to 80 or 90 fps and runs smooth. But i'm a lone fighter for playing the flat blockbuster in that way.

**Comment by cercata** [comment]
This is on the news:
https://uploadvr.com/rdr2-vr-mod-not-soon/

**Comment by VR-Punk** [comment]
In my opinion Rockstar is able to release a VR-Version, by using Nvidia Single Pass Stereo and AMD Liquid VR MultiView. Both will boost the "VR-Performance" up to 30%. I think that is the Reason, why some Triple AAA Blockbuster with official VR-Support are looking nearly just good as the Non-VR- Mode. But if the companies doesn't release VR-Versions just at the PC Release Period, we will have to wait a long time, until we get good looking Mods by great Modders like you !

**Comment by cercata** [comment]
Of course R* can do it, we have seen Robo Recall on Oculus Quest.
All we need is R* having the motivation to do it, and with 1 or 2 millions PCVR headsets, I don't think they hace it now.

But Mr Zelnick is smart guy, he's watching VR since 2016, and the 2 drawbacks he found in 2016, too expensive and too much space are not there anymore, and he's warming up with LA Noire & Borderlands, waiting for VR to grow.
One day he'll hire LukeRoss, and he'll make an official version for RDR2VR ;)

**Comment by VR-Punk** [comment]
You could start a company like Vorpx. 40 bugs for the full version. This tools 3d rendering could be based on the 3D Vision Shaderhackers, which have to be supported by donates by your company, and your turn is to develop everything else in partnership with other Modders, which would make a few bugs. Do not get me wrong, i write this text on a Ubuntu notebook, and I'm a big fan of open source, but at the moment it is more important to push the VR-Community, than to pushing linux.

**Comment by cercata** [comment]
So, did Rockstar optimize the game or what ? XD

**Comment by VR-Punk** [comment]
There is a new Vulkan 3D driver, which is compatible with RDR2. It's geometric 3D and the performance could be decent, if any shaderhacker creates a RDR2 profile for the Vulkan 3D driver. But they are not interested in such a big project beause it is a very new driver technology, and they releasing rather small projects to the Nvidia 3D Vision.

https://www.mtbs3d.com/phpbb/viewtopic.php?t=25068
https://3dsurroundgaming.com/Vk3DVision.html

**Comment by VR-Punk** [comment]
https://www.youtube.com/watch?v=xK7qvB8kdJ0&t=5s

He did it, as i suggested before ! XD

**Comment by VR-Punk** [comment]
The mod is on cs.rin.ru ! You should contact the team of cs.rin.ru !?

**Comment by LukeRoss00** [comment]
Thank you for the heads-up! I will try, but I guess that some piracy is unavoidable :-/

**Comment by cercata** [comment]
I guess it will not be a big issue, that new RVR repo will get rid of piracy ;-P
